python/lib/core/dmod/core/common/helper_functions.py

```diff
@@ -43,6 +44,34 @@ def get_mro_names(value) -> typing.List[str]:
     return list()
 
 
+def format_stack_trace(first_frame_index: int = 1) -> str:
```

Comment:
99.9% of the use cases for this will be of the form `x = format_stack_trace()`. There's only one reference to it here, but I believe that there's a more naive version floating around that is used for pretty much the exact same purpose. The variable is good for situations like handlers or being called through nested calls (such as within a base class constructor). This makes it super easy to just get the text without needing to know the implementation details.

As far as testing is concerned, you'd need to fake out the `inspect.FrameInfo` objects which would require a little bit of reverse engineering because you'd also have to come up with how you're going to fake out everything attached to it, all resulting in code that is far more complicated and prone to error than just relying on the logic within the function. `inspect` and just about _anything_ frame or traceback related generally adds a solid bit of cognitive overhead when all you want is some diagnostic text.